tests/test_type_inference/test_function_def_inference.py

```diff
@@ -54,5 +62,36 @@ def test_function_def_args_simple_return(function_name, arguments):
         assert Callable[actual_arg_types, actual_return_type] == Callable[expected_arg_types, expected_return_type]
 
 
+@given(cs.valid_identifier(), hs.lists(cs.valid_identifier(), min_size=2), hs.lists(cs.annotation, min_size=3))
+def test_functiondef_annotated_simple_return(function_name, arguments, annotations):
+    """Test whether type annotations are set properly for a FunctionDef node representing a function definition
+    with type annotations."""
+    # TODO: 2 test cases?... one for args, one for return?
+    assume(function_name not in arguments and len(arguments) == len(annotations) - 1)
+    [assume(annotation is not None) for annotation in annotations]
+    program = _parse_to_annotated_function(function_name
+                                           , arguments, (arguments[0]), annotations)
+    module, inferer = cs._parse_text(program)
+    functiondef_node = next(module.nodes_of_class(astroid.FunctionDef))
+    # arguments and annotations are not changing, so test this once.
+    for i in range(len(functiondef_node.args.annotations)):
+        arg_name = functiondef_node.args.args[i].name
+        actual_type = inferer.type_constraints.lookup_concrete(functiondef_node.type_environment.lookup_in_env(arg_name))
+        assert actual_type.__name__ == functiondef_node.args.annotations[i].name
+    # test return type annotations
+    for i in range(len(arguments)):
+        annotations[-1] = annotations[i]
+        program = _parse_to_annotated_function(function_name
+                                               , arguments, (arguments[i]), annotations)
+        module, inferer = cs._parse_text(program)
+        functiondef_node = next(module.nodes_of_class(astroid.FunctionDef))
+        expected_arg_type_vars = [functiondef_node.type_environment.lookup_in_env(argument) for argument in arguments]
+        expected_arg_types = [inferer.type_constraints.lookup_concrete(type_var) for type_var in expected_arg_type_vars]
+        function_type_var = module.type_environment.lookup_in_env(function_name)
+        function_type = inferer.type_constraints.lookup_concrete(function_type_var)
+        actual_arg_types, actual_return_type = inferer.type_constraints.types_in_callable(function_type)
+        assert expected_arg_types == actual_arg_types and functiondef_node.returns.name == actual_return_type.__name__
```

Comment:
There is no reason to compare names here. This is related to the previous comment about types vs. strings.